### PR TITLE
Make LocalSyslog only available under Linux

### LINKED ISSUE #58: Should LocalSyslogService.cs be automatically disabled on non-Linux systems?
[LocalSyslogService.cs](https://github.com/IonxSolutions/serilog-sinks-syslog/blob/master/src/Serilog.Sinks.Syslog/Sinks/LocalSyslogService.cs)

LocalSyslogService.cs uses a Linux-specific api (imported from the libc library). If LocalSys is enabled on windows, can it be automatically turned into a nop operation?

All the sinks of our serilog are defined in the configuration file. I don't know how to dynamically specify the runtime conditions in the configuration file to automatically disable the LocalSys sink on non-linux systems.

thanks

### PR DESCRIPTION
**Pull request by pedoc**
fixed #58

### CODE REVIEW

File: src/Serilog.Sinks.Syslog/Serilog.Sinks.Syslog.csproj [hunk 1]
```diff
@@ -28,9 +28,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
-    <PackageReference Include="Serilog" Version="2.5.0" />
-    <PackageReference Include="Serilog.Sinks.PeriodicBatching" Version="2.3.0" />
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" PrivateAssets="All" />
+    <PackageReference Include="Serilog" Version="2.12.0" />
```
[the remark this answers is not in the document]
Comment (pedoc) [reply src/Serilog.Sinks.Syslog/Serilog.Sinks.Syslog.csproj:32]: Yes, upgrading dependencies brings some breaking changes.
Let me explain why I want to upgrade the dependency

In our project, we use both Serilog.Console and Serilog.File, which are very commonly used libraries. But each library has dependencies on different versions of Serilog, which may cause some unexpected errors, and Difficult to detect. So I upgraded it to the latest version, and made the dependencies of these libraries on Serilog consistent.

Regarding the issue of upgrading dependencies, my personal opinion is that if this is indeed a breaking change, maybe we can release a larger version number, such as 2.x->3.x, which is also a common approach, what do you think?

If you really think that dependencies should not be upgraded, please let me know and I will undo this change.
Thanks

File: src/Serilog.Sinks.Syslog/Serilog.Sinks.Syslog.csproj [hunk 2]
```diff
@@ -28,9 +28,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
-    <PackageReference Include="Serilog" Version="2.5.0" />
-    <PackageReference Include="Serilog.Sinks.PeriodicBatching" Version="2.3.0" />
```
[the remark this answers is not in the document]
Comment (pedoc) [reply src/Serilog.Sinks.Syslog/Serilog.Sinks.Syslog.csproj:33]: For ``Serilog.Sinks.PeriodicBatching`` Dependencies I upgraded without finding any obvious exceptions. As with the previous question, if you think these dependencies should not be upgraded, I will revoke them.

File: src/Serilog.Sinks.Syslog/Serilog.Sinks.Syslog.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (colin-anderson) [reply src/Serilog.Sinks.Syslog/Serilog.Sinks.Syslog.csproj:32]: I prefer to deal with this by not relying on a transitive dependency; instead, taking a direct dependency on the `Serilog` nuget, loading the version I actually want.

Either way, it's something not really related to this PR, so I'd prefer if you removed it please 🙏

File: src/Serilog.Sinks.Syslog/Serilog.Sinks.Syslog.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (pedoc) [reply src/Serilog.Sinks.Syslog/Serilog.Sinks.Syslog.csproj]: > Yes, please do remove this change, then I'll get this approved and a new NuGet out hopefully tomorrow 👍

The dependency upgrade has been removed according to your request. In order to keep the commit log clean, I used reset 😄

**Comment by colin-anderson** [comment]
Thanks for this, I'll try to get a new NuGet out tomorrow 👍

**Comment by colin-anderson** [comment]
v2.0.8 is now released on NuGet, thanks again for this 🙏

**Comment by pedoc** [comment]
> v2.0.8 is now released on NuGet, thanks again for this 🙏

You're welcome, and thank you for such a great library that makes my job easier 😄

**Pull request merged**